### PR TITLE
Update serp_events derived table

### PR DESCRIPTION
**Pull request by dzeber**
This adds a new derived table verison for `serp_events` as discussed in #5296. View to be updated in a separate PR once this table has been built.

Summary of changes:
- 1 row per impression ID with ad impression & engagement info grouped into arrays (v1 has multiple rows per impression ID)
- Count all engagement actions rather than specific ones. New engagement actions added to the events later on will automatically get included in the table.
- Mapping logic moved out of derived table
- Change to the logic in counting visible ads following updated discussions with Eng.
    - previously, we used `num_ads_showing` to count visible ads, depending on the ad blocker inference logic.
    - we now use reported `ads_visible` as the count of visible ads, and `num_ads_showing` column is removed
    - `_reported` suffix dropped from ad impression counts
    - change naming of `_hidden` to `_blocked`
- Change naming in table columns from 'ad impressions' to 'component/element impressions' to reflect broader usage of `ad_impression` event
- Add a few new fields from `serp.impression` events
- Incorporate unmerged changes from #5189 
- Add description in metadata.yaml

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3510)

### CODE REVIEW

File: sql_generators/serp_events_v2/templates/view.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+--- User-facing view. Generated via sql_generators.active_users.
+CREATE OR REPLACE VIEW
+  `{{ project_id }}.{{ app_name }}.serp_events_v2`
```
Comment (alekhyamoz) [inline sql_generators/serp_events_v2/templates/view.sql:3]: You want to create a separate view as well and then later repoint?

**Review by alekhyamoz — CHANGES_REQUESTED** [review]

File: sql_generators/serp_events_v2/templates/view.sql (hunk 1 above)
Comment (dzeber) [reply sql_generators/serp_events_v2/templates/view.sql:3]: You're right, not sure why I added this. I'll remove.

Comment (dzeber) [reply sql_generators/serp_events_v2/templates/view.sql:3]: You're right, not sure why I added this. I'll remove.

**Review by alekhyamoz — APPROVED** [review]
LGTM

**Pull request merged**